pkg/storage/objects/store.go

```diff
@@ -66,8 +73,22 @@ func (o *objStore) Meta(ctx context.Context, path paths.Path) (meta Meta,
 func (o *objStore) Get(ctx context.Context, path paths.Path) (
 	rr ranger.RangeCloser, meta Meta, err error) {
 	defer mon.Task()(&ctx)(&err)
+
 	rr, m, err := o.s.Get(ctx, path)
-	return rr, convertMeta(m), err
+
+	encKey := sha256.Sum256([]byte(*key))
```

Comment:
instead, perhaps the constructor could assert that the key is the right length? someone else's problem to generate a valid encryption key